server/types/bool.go

```diff
@@ -143,6 +159,10 @@ func (b BoolType) IoOutput(output any) (string, error) {
 	}
 }
 
+func (b BoolType) IsPreferredType() bool {
```

Comment:
I'd throw a comment on these functions too. Looks like it's missing across all of the types.